src/System.Linq.Expressions/src/System/Dynamic/Utils/TypeUtils.cs

```diff
@@ -730,5 +730,13 @@ internal static bool IsFloatingPoint(Type type)
             }
         }
 #endif 
+
+        public static bool IsVector(this Type type)
+        {
+            // Unfortunately, the IsSzArray property of System.Type is inaccessible to us,
+            // but the ToString representation of the type is sufficient to differentiate
+            // a vector (X[]) from a multi-dimensional array of rank 1 (X[*]).
+            return type.GetArrayRank() == 1 && type.ToString().EndsWith("[]");
```

Comment:
Would checking that the lower bound is zero be worth doing? I imagine it's faster than string comparison (but haven't checked), but then it's probably optimising the less-common case, so perhaps not worth it.